src/winapp-npm/dependency-utils.js

```diff
@@ -0,0 +1,371 @@
+    if (fsSync.existsSync(vswherePath)) {
+      vswhereCmd = vswherePath;
+    }
+    
+    try {
+      // Use vswhere to find Visual Studio 2022, 2026, or later with BuildTools
+      const output = execSync(`"${vswhereCmd}" -products * -requires Microsoft.VisualStudio.Workload.NativeDesktop -property installationPath`, { 
+        encoding: 'utf8',
+        stdio: 'pipe'
+      }).trim();
+      
+      // If we got any output, VS with the required workload is installed
+      if (output && output.length > 0) {
+        return true;
+      }
+    } catch (error) {
+      // vswhere not found or query failed, fall back to cl.exe check
+    }
+  } catch (error) {
+    return false;
+  }
+  return false;
+}
+
+/**
+ * Get the winget command line for installing Visual Studio Build Tools
+ * @returns {string} The winget command line
+ */
+function getVisualStudioBuildToolsWingetCommand() {
+  const components = [
+    'Microsoft.VisualStudio.Workload.NativeDesktop'
+  ];
+  
+  const addFlags = components.map(c => `--add ${c}`).join(' ');
+  return `winget install --id Microsoft.VisualStudio.Community --source winget --force --override "${addFlags} --passive --includeRecommended --wait"`;
+}
+
+/**
+ * Install Visual Studio Build Tools using winget
+ * @returns {Promise<boolean>} true if successful, false otherwise
+ */
+function installVisualStudioBuildTools() {
+  return new Promise(resolve => {
+    const { spawn } = require('child_process');
+    
+    // Use winget to install Visual Studio Community with Native Desktop workload
+    const command = getVisualStudioBuildToolsWingetCommand();
+    const winget = spawn(command, {
+      stdio: 'inherit',
+      shell: true
+    });
+    
+    winget.on('close', (code) => {
+      resolve(code === 0);
+    });
+    
+    winget.on('error', (err) => {
+      console.error(`Error running winget: ${err.message}`);
+      resolve(false);
+    });
+  });
+}
+
+/**
+ * Check for Visual Studio Build Tools and offer to install if missing
+ * @param {boolean} verbose - Enable verbose logging
+ * @returns {Promise<boolean>} true if something was installed, false otherwise
+ */
+async function checkAndInstallVisualStudioBuildTools(verbose = false) {
+  const hasVisualStudioBuildTools = checkVisualStudioBuildTools();
+  if (!hasVisualStudioBuildTools) {
+    // Check if we're in an interactive terminal
+    if (process.stdin.isTTY) {
+      const readline = require('readline');
+      const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout
+      });
+      
+      return new Promise((resolve) => {
+        rl.question('❓ Visual Studio Build Tools are not installed - install using winget (user interaction may be required)? (y/N): ', async (answer) => {
+          rl.close();
+          
+          if (answer.toLowerCase() === 'y') {
+            console.log('');
+            console.log(`Installing with \`${getVisualStudioBuildToolsWingetCommand()}\``);
```

Comment:
Given that there are multiple versions of the VS installer (and this is assuming Community), I would not run the command without explicitly saying that we will run install the Community version before asking for the confirmation, not after.